tests/pass-dep/concurrency/apple-futex.rs

```diff
@@ -0,0 +1,211 @@
+//@only-target: darwin
+
+use std::time::{Duration, Instant};
+use std::{io, ptr, thread};
+
+fn wake_nobody() {
+    let futex = 0;
+
+    // Wake 1 waiter. Expect zero waiters woken up, as nobody is waiting.
+    unsafe {
+        assert_eq!(
+            libc::os_sync_wake_by_address_any(
+                ptr::from_ref(&futex).cast_mut().cast(),
+                size_of::<i32>(),
+                libc::OS_SYNC_WAKE_BY_ADDRESS_NONE
+            ),
+            -1
+        );
+        assert_eq!(io::Error::last_os_error().raw_os_error().unwrap(), libc::ENOENT);
+    }
+}
+
+fn wake_dangling() {
+    let futex = Box::new(0);
+    let ptr = ptr::from_ref(&futex).cast_mut().cast();
+    drop(futex);
+
+    // Expect error since this is now "unmapped" memory.
+    unsafe {
+        assert_eq!(
+            libc::os_sync_wake_by_address_any(
+                ptr,
+                size_of::<i32>(),
+                libc::OS_SYNC_WAKE_BY_ADDRESS_NONE
+            ),
+            -1
+        );
+        assert_eq!(io::Error::last_os_error().raw_os_error().unwrap(), libc::ENOENT);
```

Comment:
So this is the same error as "nobody is waiting"? Interesting.

Reply:
Yeah but none of the other systems return an error when nobody is waiting, AFAIK.

Reply:
True, but Apple explicitly documents `ENOENT` to be the error code returned in this case...

Reply:
Yeah, it's fine. I was just somewhat surprised here that "no waiter" and "no memory" are the same error code, but I guess if there's no memory then there's definitely nobody waiting so this makes some sense.